server/src/main/java/org/elasticsearch/cluster/metadata/IndexReshardingState.java

```diff
@@ -380,6 +380,10 @@ public boolean inProgress() {
             return false;
         }
 
+        public boolean allTargetsSplit() {
+            return Arrays.stream(targetShards).allMatch(target -> target == TargetShardState.SPLIT);
+        }
```

Comment:
I think this is fine, but I'm not sure it's necessary either. I am not sure whether we'll end up having a strong use case for this particular state of a split in the end, and its caller in the other review probably would have been fine inlining this.